sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxClientTests.cs

```diff
@@ -0,0 +1,406 @@
+               DevBoxName,
+               userId: TestEnvironment.UserId);
+
+            JsonElement devBoxResponseData = JsonDocument.Parse(devBoxResponse.ContentStream).RootElement;
+
+            if (!devBoxResponseData.TryGetProperty("name", out var devBoxNameJson))
+            {
+                FailDueToMissingProperty("name");
+            }
+
+            string devBoxName = devBoxNameJson.ToString();
+            Assert.AreEqual(devBoxName, DevBoxName);
+        }
+
+        [RecordedTest]
+        public async Task GetDevBoxesSucceeds()
+        {
+            int numberOfReturnedDevBoxes = 0;
+
+            await foreach (BinaryData devBoxData in _devBoxesClient.GetDevBoxesAsync(TestEnvironment.ProjectName))
+            {
+                numberOfReturnedDevBoxes++;
+                JsonElement devBoxResponseData = JsonDocument.Parse(devBoxData.ToStream()).RootElement;
+
+                if (!devBoxResponseData.TryGetProperty("name", out var devBoxNameJson))
+                {
+                    FailDueToMissingProperty("name");
+                }
+
+                string devBoxName = devBoxNameJson.ToString();
+                Assert.AreEqual(devBoxName, DevBoxName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedDevBoxes);
+        }
+
+        [RecordedTest]
+        public async Task GetAllDevBoxesSucceeds()
+        {
+            int numberOfReturnedDevBoxes = 0;
+
+            await foreach (BinaryData devBoxData in _devBoxesClient.GetAllDevBoxesAsync())
+            {
+                numberOfReturnedDevBoxes++;
+                JsonElement devBoxResponseData = JsonDocument.Parse(devBoxData.ToStream()).RootElement;
+
+                if (!devBoxResponseData.TryGetProperty("name", out var devBoxNameJson))
+                {
+                    FailDueToMissingProperty("name");
+                }
+
+                string devBoxName = devBoxNameJson.ToString();
+                Assert.AreEqual(devBoxName, DevBoxName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedDevBoxes);
+        }
+
+        [RecordedTest]
+        public async Task GetAllDevBoxesByUserSucceeds()
+        {
+            int numberOfReturnedDevBoxes = 0;
+
+            await foreach (BinaryData devBoxData in _devBoxesClient.GetAllDevBoxesByUserAsync(userId: TestEnvironment.UserId))
+            {
+                numberOfReturnedDevBoxes++;
+                JsonElement devBoxResponseData = JsonDocument.Parse(devBoxData.ToStream()).RootElement;
+
+                if (!devBoxResponseData.TryGetProperty("name", out var devBoxNameJson))
+                {
+                    FailDueToMissingProperty("name");
+                }
+
+                string devBoxName = devBoxNameJson.ToString();
+                Assert.AreEqual(devBoxName, DevBoxName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedDevBoxes);
+        }
+
+        [RecordedTest]
+        public async Task GetPoolSucceeds()
```

Comment:
Is setup running for these non-dev box tests as well? If so, may be worthwhile to split tests not requiring that dev box setup/teardown into a separate test class. Not strictly necessary at the moment, since we can mostly rely on recordings, but a good fast-follow if live test performance/reliability is getting impacted by dev box creation.

Reply:
Added the check: https://github.com/Azure/azure-sdk-for-net/pull/36245/commits/7bf2bf9b6d5e61f810fb6e614a8d70abdf8e14f7

Reply:
Agree.